logstash-output-elasticsearch.gemspec

```diff
@@ -25,6 +25,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'ftw', '~> 0.0.42'
   s.add_development_dependency 'addressable', "~> 2.3.0" # used by FTW. V 2.5.0 is ruby 2.0 only.
   s.add_development_dependency 'logstash-codec-plain'
+  s.add_development_dependency 'json' # used by spec/unit/outputs/elasticsearch/http_client/pool_spec.rb
```

Comment:
Once again, forgive my noobishness, but I think gemspec allows just `add_development_dependency` and `add_runtime_dependency`. If it's only required by tests, shouldn't it be a development dep? Otherwise, it would leak json into any downstream projects.

While I was at it, i thought it would be nifty to make sure json is actually required. I applied this diff:
```diff
diff --git a/logstash-output-elasticsearch.gemspec b/logstash-output-elasticsearch.gemspec
index 5c045c8..6f0b38b 100644
--- a/logstash-output-elasticsearch.gemspec
+++ b/logstash-output-elasticsearch.gemspec
@@ -25,7 +25,6 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'ftw', '~> 0.0.42'
   s.add_development_dependency 'addressable', "~> 2.3.0" # used by FTW. V 2.5.0 is ruby 2.0 only.
   s.add_development_dependency 'logstash-codec-plain'
-  s.add_development_dependency 'json' # used by spec/unit/outputs/elasticsearch/http_client/pool_spec.rb
 
   if RUBY_PLATFORM == 'java'
     s.platform = RUBY_PLATFORM
diff --git a/spec/integration/outputs/sniffer_spec.rb b/spec/integration/outputs/sniffer_spec.rb
index f3d5685..c54591b 100644
--- a/spec/integration/outputs/sniffer_spec.rb
+++ b/spec/integration/outputs/sniffer_spec.rb
@@ -1,7 +1,6 @@
 require "logstash/devutils/rspec/spec_helper"
 require_relative "../../../spec/es_spec_helper"
 require "logstash/outputs/elasticsearch/http_client"
-require "json"
 
 describe "pool sniffer", :integration => true do
   let(:logger) { Cabin::Channel.get }
diff --git a/spec/unit/outputs/elasticsearch/http_client/pool_spec.rb b/spec/unit/outputs/elasticsearch/http_client/pool_spec.rb
index 3406623..5bfe2e6 100644
--- a/spec/unit/outputs/elasticsearch/http_client/pool_spec.rb
+++ b/spec/unit/outputs/elasticsearch/http_client/pool_spec.rb
@@ -1,6 +1,5 @@
 require "logstash/devutils/rspec/spec_helper"
 require "logstash/outputs/elasticsearch/http_client"
-require "json"
 
 describe LogStash::Outputs::ElasticSearch::HttpClient::Pool do
   let(:logger) { Cabin::Channel.get }
diff --git a/spec/unit/outputs/elasticsearch/http_client_spec.rb b/spec/unit/outputs/elasticsearch/http_client_spec.rb
index 3ade145..020b31e 100644
--- a/spec/unit/outputs/elasticsearch/http_client_spec.rb
+++ b/spec/unit/outputs/elasticsearch/http_client_spec.rb
@@ -125,7 +125,6 @@ describe LogStash::Outputs::ElasticSearch::HttpClient do
     subject { described_class.new(base_options) }
 
     context "when items key is available" do
-      require "json"
       let(:bulk_response) {
         LogStash::Json.load ('[{
           "items": [{
@@ -149,7 +148,6 @@ describe LogStash::Outputs::ElasticSearch::HttpClient do
     end
 
     context "when items key is not available" do
-      require "json"
       let(:bulk_response) {
         JSON.parse ('[{
           "took": 4,
diff --git a/spec/unit/outputs/elasticsearch/template_manager_spec.rb b/spec/unit/outputs/elasticsearch/template_manager_spec.rb
index d63fb05..ba8825c 100644
--- a/spec/unit/outputs/elasticsearch/template_manager_spec.rb
+++ b/spec/unit/outputs/elasticsearch/template_manager_spec.rb
@@ -1,7 +1,6 @@
 require "logstash/devutils/rspec/spec_helper"
 require "logstash/outputs/elasticsearch/http_client"
 require "java"
-require "json"
 
 describe LogStash::Outputs::ElasticSearch::TemplateManager do
```

And running the (unit) tests gives the following failures:

```
Failures:

  1) LogStash::Outputs::ElasticSearch::TemplateManager.get_es_major_version elasticsearch 2.x detects major version is 2
     Failure/Error: allow(client).to receive(:get_version).and_return(JSON.parse(es_2x_version))

     NameError:
       uninitialized constant JSON
     # ./spec/unit/outputs/elasticsearch/template_manager_spec.rb:22:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'

  2) LogStash::Outputs::ElasticSearch::TemplateManager.get_es_major_version elasticsearch 5.x detects major version is 5
     Failure/Error: allow(client).to receive(:get_version).and_return(JSON.parse(es_5x_version))

     NameError:
       uninitialized constant JSON
     # ./spec/unit/outputs/elasticsearch/template_manager_spec.rb:30:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'

  3) LogStash::Outputs::ElasticSearch::TemplateManager.get_es_major_version elasticsearch 1.x detects major version is 1
     Failure/Error: allow(client).to receive(:get_version).and_return(JSON.parse(es_1x_version))

     NameError:
       uninitialized constant JSON
     # ./spec/unit/outputs/elasticsearch/template_manager_spec.rb:14:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'

  4) LogStash::Outputs::ElasticSearch::HttpClient join_bulk_responses when items key is not available should be handled properly
     Failure/Error:
       JSON.parse ('[{
         "took": 4,
         "errors": false
       }]')

     NameError:
       uninitialized constant JSON
     # ./spec/unit/outputs/elasticsearch/http_client_spec.rb:152:in `bulk_response'
     # ./spec/unit/outputs/elasticsearch/http_client_spec.rb:158:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'

```

So I'm thinking we do need this dep, and it should be a development dep. Have I missed something?